sonar-batch/src/main/java/org/sonar/batch/mediumtest/TaskResult.java

```diff
@@ -250,13 +240,12 @@ private static TextRange toRange(InputFile file, Range reportRange) {
   @CheckForNull
   public BatchReport.Coverage coverageFor(InputFile file, int line) {
     int ref = reportComponents.get(((DefaultInputFile) file).key()).getRef();
-    try (InputStream inputStream = FileUtils.openInputStream(getReportReader().readComponentCoverage(ref))) {
-      BatchReport.Coverage coverage = BatchReport.Coverage.PARSER.parseDelimitedFrom(inputStream);
-      while (coverage != null) {
+    try (CloseableIterator<BatchReport.Coverage> it = getReportReader().readComponentCoverage(ref)) {
+      while (it.hasNext()) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'it' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2259)